find_cmake_package_name.py

```diff
@@ -78,40 +74,91 @@ def find_path(conan_root, package_info):
+
+
+def find_configuration(package_path):
     """Find CMake configuration files.
 
     package_path -- The directory path to start searching from.
     """
 
-    config_suffix = re.compile('(C|-c)onfig.cmake$')
-    f = list()
-    for _, _, files in walk(package_path):
-        f += list(filter(lambda file: config_suffix.search(file), files))
-    f = list(dict.fromkeys(f))
-    return list(map(lambda i: config_suffix.sub('', i), f))
+    configurations = list()
 
+    for root, _, files in walk(package_path):
+        configurations += list(
+            filter(lambda file: config_suffix.search(file), files))
 
-def main():
-    package_ref, conan_root = parse_args()
-    package_path = find_path(conan_root, split_package_ref(package_ref))
+    configurations = list(
+        map(lambda i: config_suffix.sub('', i), dict.fromkeys(configurations)))
+
+    if len(configurations) == 0:
+        raise RuntimeError("Could not find configuration")
 
-    configurations = find_configurations(package_path)
+    return configurations[0]
+
+
+target_suffix = re.compile('Targets.cmake$')
+
+
+def find_linkage(package_path, package_name):
+    """Find CMake configuration files.
 
-    for i in configurations:
-        print(i)
-    exit(len(configurations) == 0)
+    package_path -- The directory path to start searching from.
+    package_name -- The name of the package
+    """
+
+    for root, _, files in walk(package_path):
+        for cmake_file in filter(
+                lambda f: config_suffix.search(f) or target_suffix.search(f),
+                files):
+            with open(f"{root}/{cmake_file}") as to_read:
+                data = to_read.read()
+                if "::" in data:
+                    result = re.search("([A-Za-z0-9_-]+)::[A-Za-z0-9_-]+",
+                                       data)
+                    if result[1] == package_name:
+                        return result[0]
+    return package_name
+
+
+def print_template(path, substitutions):
+    """
+        A crude way to handle string templating until I work out string template engines in Python.
+
+        path -- The path to a string template file.
+        substitutions -- A dictionary containing a mapping for string templates.
+    """
+    with open(path) as file:
+        template = file.read()
+        for key, value in substitutions.items():
+            template = re.sub(key, value, template)
+        print(template)
+
+
+def main():
+    try:
+        package_ref, conan_root = parse_args()
+        package_path = find_path(conan_root, split_package_ref(package_ref))
+
+        package_name = find_configuration(package_path)
+        package_link = find_linkage(package_path, package_name)
+
+        root = "config/python"
+        print_template(f"{root}/find_package.st", {"__name__": package_name})
+        print_template(f"{root}/link.st", {"__link__": package_link})
+    except RuntimeError as error:
+        print(f"{error} for {package_ref}.")
```

Comment:
If you plan on having lots of output and need to highlight certain things, [crayons](https://github.com/kennethreitz/crayons) is very simple to add to a project.

Reply:
Can I use this for my `*.st` files? There's some content in there that I'd like to highlight.

Reply:
Crayons was a huge hit, once I learnt how to use it.